analyzers/src/SonarAnalyzer.VisualBasic/Rules/MultipleVariableDeclarationCodeFix.cs

```diff
@@ -18,79 +18,50 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.Formatting;
-using Microsoft.CodeAnalysis.VisualBasic;
-using Microsoft.CodeAnalysis.VisualBasic.Syntax;
 
 namespace SonarAnalyzer.Rules.VisualBasic
 {
     [ExportCodeFixProvider(LanguageNames.VisualBasic)]
     public sealed class MultipleVariableDeclarationCodeFix : MultipleVariableDeclarationCodeFixBase
     {
-        protected override SyntaxNode CalculateNewRoot(SyntaxNode root, SyntaxNode node)
-        {
-            if (!(node is ModifiedIdentifierSyntax identifier))
-            {
-                return root;
-            }
+        protected override SyntaxNode CalculateNewRoot(SyntaxNode root, SyntaxNode node) =>
+            node is ModifiedIdentifierSyntax { Parent: VariableDeclaratorSyntax declarator }
+                ? root.ReplaceNode(declarator.Parent, CreateNewNodes(declarator))
+                : root;
 
-            if (!(identifier.Parent is VariableDeclaratorSyntax declarator))
+        private static IEnumerable<SyntaxNode> CreateNewNodes(VariableDeclaratorSyntax declarator) =>
+            declarator.Parent switch
             {
-                return root;
-            }
+                FieldDeclarationSyntax fieldDeclaration => CreateNewNodes(fieldDeclaration),
+                LocalDeclarationStatementSyntax localDeclaration => CreateNewNodes(localDeclaration),
+                _ => Enumerable.Empty<SyntaxNode>()
```

Comment:
This `Empty` changes the logic. 
Previously, no change was done. Root with no replacement was returned.
Now, we remove the declaration completely. 

This is a bit of a workaround, but the code should not be reachable and testable anyway. So the replace for th same node doesn't cost us anything
```suggestion
                _ => declaration.Parent
```